### PR TITLE
do not override existing environment variables

### LINKED ISSUE #13: Do not override GOOGLE_APPLICATION_CREDENTIALS env var if it's already set in a pod.
Related: https://github.com/kubernetes/minikube/issues/11563

### PR DESCRIPTION
**Pull request by sharifelgamal**
Fixes #13 

Now, the webhook will only set an env var if it isn't already in the pod

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -90,11 +90,16 @@ func mutateHandler(w http.ResponseWriter, r *http.Request) {
...
 
 	var patch []patchOperation
+	var envVars []corev1.EnvVar
+
+	needsCreds := needsEnvVar(pod.Spec.Containers[0], "GOOGLE_APPLICATION_CREDENTIALS")
 
 	// Explicitly and silently exclude the kube-system namespace
 	if pod.ObjectMeta.Namespace != metav1.NamespaceSystem {
+		var v corev1.Volume
+		var mount corev1.VolumeMount
```
Comment (spowelljr) [inline server.go:100]: I'm not against this in anyway, I'm purely just curious if there's a specific reason why you did this

Comment (sharifelgamal) [reply server.go:100]: ah, i had a different refactor that required this, but then i removed it. fixed.

File: server.go [hunk 2]
```diff
@@ -118,58 +121,68 @@ func mutateHandler(w http.ResponseWriter, r *http.Request) {
...
-					Op:    "add",
-					Path:  fmt.Sprintf("/spec/containers/%d/env", i),
-					Value: append(c.Env, envVars...),
-				})
+		if needsCreds {
+			patch = append(patch, patchOperation{
+				Op:    "add",
+				Path:  "/spec/volumes",
+				Value: append(pod.Spec.Volumes, v),
+			})
```
Comment (spowelljr) [inline server.go:153]: Could you combine this `if` with the identical one on L124?

Comment (sharifelgamal) [reply server.go]: good call, done

**Review by spowelljr — APPROVED** [review]

**Pull request merged**